### PR TITLE
fixes #4425 feat(nimbus): add end experiment UI

### LINKED ISSUE #4425: Add end experiment button to UI
Once we have the mutation to end an experiment and the ending state is exposed in the get experiment by slug query, we can add the button to the UI.  Clicking it should invoke the mutation and re read the experiment which should now indicate that its ending is in progress so we can then mark in the UI somewhere that it's in the progress of ending and the button is disabled.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/EXP-832)
┆Issue Number: EXP-832

### PR DESCRIPTION
**Pull request by jodyheavener**
This PR adds the UI to request the end of an Experiment, including the initial button to end, the confirmation dialogue, and the "is ending" dialogue.

Storybook:

- [Summary page with LIVE experiment](https://storage.googleapis.com/mozilla-storybooks-experimenter/commits/5aee79269523d72af18e100fec17c2299b542a05/nimbus-ui/index.html?path=/story/components-summary--live-status)
- [Summary page with ENDING experiment](https://storage.googleapis.com/mozilla-storybooks-experimenter/commits/5aee79269523d72af18e100fec17c2299b542a05/nimbus-ui/index.html?path=/story/components-summary--end-requested)

### CODE REVIEW

**Comment by jodyheavener** [comment]
@AnaMedinac @k88hudson for your consideration

**Comment by k88hudson** [comment]
LGTM, `Users are still seeing the experiment until ending has been approved in Remote Settings. Once it has been approved, your experiment will be in “Complete” state.` is a bit wordy but descriptive

**Comment by AnaMedinac** [comment]
Wondering if we can shorten the dialog for waiting for approval to be:
"Users will no longer see the experiment once ending is approved in Remote Settings, and is in "Complete" state".

**Comment by jaredlockhart** [comment]
I noticed that on some pages our 'next step' controls are at the bottom (like to launch) and on some they're at the top (like to end).  Maybe that should be more consistent?

**Comment by jodyheavener** [comment]
@jaredlockhart do you mean the request review one? that's going to the top [as well](*https://storage.googleapis.com/mozilla-storybooks-experimenter/commits/5aee79269523d72af18e100fec17c2299b542a05/nimbus-ui/index.html?path=/story/pages-requestreview-exp-866--draft-status)

**Comment by jaredlockhart** [comment]
> @jaredlockhart do you mean the launch to preview one? that's going to the top [as well](https://storage.googleapis.com/mozilla-storybooks-experimenter/commits/5aee79269523d72af18e100fec17c2299b542a05/nimbus-ui/index.html?path=/story/pages-requestreview-exp-866--draft-status)

Oh right so that replaces the current launch one that's at the bottom now, so they'll all be at the top?  Oh good then that's good never mind me.

File: app/experimenter/nimbus-ui/src/components/Summary/EndExperiment/index.tsx
Comment (lmorchard) [inline app/experimenter/nimbus-ui/src/components/Summary/EndExperiment/index.tsx:39]: Do we maybe need a "loading" state here while this resolves? Probably not a big deal since this mutation should be idempotent

File: app/experimenter/nimbus-ui/src/components/Summary/EndExperiment/index.tsx
Comment (lmorchard) [inline app/experimenter/nimbus-ui/src/components/Summary/EndExperiment/index.tsx:116]: I think we should check for `!status.complete && isEnding` here - this message should disappear after the pending end has been approved

**Review by lmorchard — CHANGES_REQUESTED** [review]
Ran this through its paces locally - looks good except for the final complete state, when the pending message should no longer appear

File: app/experimenter/nimbus-ui/src/components/Summary/EndExperiment/index.tsx
Comment (jodyheavener) [reply app/experimenter/nimbus-ui/src/components/Summary/EndExperiment/index.tsx:39]: Both buttons are [disabled](https://github.com/mozilla/experimenter/pull/4545/files#diff-a34086e9954d0954013483aa493baa2be3f4c9f94eadf5ca88198813109060daR97) while the mutation does its thing - that what you mean?

File: app/experimenter/nimbus-ui/src/components/Summary/EndExperiment/index.tsx
Comment (jodyheavener) [reply app/experimenter/nimbus-ui/src/components/Summary/EndExperiment/index.tsx:116]: Oh interesting I assumed `isEndRequested` would return to false after it was moved into a complete state.

Comment (jodyheavener) [reply app/experimenter/nimbus-ui/src/components/Summary/EndExperiment/index.tsx]: (I moved the status.live check up to <Summary> btw)

File: app/experimenter/nimbus-ui/src/components/Summary/EndExperiment/index.tsx
Comment (lmorchard) [reply app/experimenter/nimbus-ui/src/components/Summary/EndExperiment/index.tsx:39]: Oh, derp, I missed that!

File: app/experimenter/nimbus-ui/src/lib/experiment.ts
Comment (lmorchard) [inline app/experimenter/nimbus-ui/src/lib/experiment.ts:13]: I see the `status.live` check in the `Summary` component, though this should probably be `status === NimbusExperimentStatus.LIVE && !!experiment?.isEndRequested` since that's more exactly what the state will be. But, nothing functional here would change with that, just might clarify things vaguely in the future.

**Review by lmorchard — APPROVED** [review]

**Comment by lmorchard** [comment]
Even with that last comment about status logic, I'd say :+1: anyway

File: app/experimenter/nimbus-ui/src/lib/experiment.ts
Comment (jodyheavener) [reply app/experimenter/nimbus-ui/src/lib/experiment.ts:13]: Fair point

**Pull request merged**
